src/main/scala/org/hammerlab/guacamole/reads/ReadPair.scala

```diff
@@ -0,0 +1,41 @@
+package org.hammerlab.guacamole.reads
+
+import htsjdk.samtools.SAMRecord
+
+/**
+ * *
+ * @param read Unmapped or MappedRead base read
+ * @param isFirstInPair Whether the read is earlier that the the mate read
+ * @param mateAlignmentProperties Alignment location of the mate if it the mate is aligned
+ * @tparam T UnmappedRead or MappedRead
+ */
+case class ReadPair[+T <: Read](read: T,
```

Comment:
Yes, I agree `PairedRead` is a better name. It is a `Mapped/UnmappedRead` that is a paired end read and the alignment information of the mate.